From ae4692c1c007a19ff577c3939531160b9d8ba756 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Mon, 22 Oct 2018 16:49:39 +0800
Subject: [PATCH 5003/5242] MLK-20014: ISI: initialize SCL_IMG_CFG to source
 image width and height

commit  6343424babf861da68e84754f2e3a0e3e988cac4 from
https://source.codeaurora.org/external/imx/linux-imx.git

Initialize SCL_IMG_CFG to source image width and height when
scale disabled

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit 506b32bb9ad6a3f596510e401b0e8fa16ca8b659)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-isi-hw.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/platform/imx8/mxc-isi-hw.c b/drivers/media/platform/imx8/mxc-isi-hw.c
index 0d100cb..6f0d7be 100644
--- a/drivers/media/platform/imx8/mxc-isi-hw.c
+++ b/drivers/media/platform/imx8/mxc-isi-hw.c
@@ -691,6 +691,10 @@ void mxc_isi_m2m_channel_config(struct mxc_isi_dev *mxc_isi)
 			mxc_isi->m2m.alphaen << CHNL_IMG_CTRL_GBL_ALPHA_EN_OFFSET);
 	writel(reg, mxc_isi->regs + CHNL_IMG_CTRL);
 
+	/* scale size need to equal input size when scaling disabled*/
+	reg = src_f->o_width | (src_f->o_height << CHNL_IMG_CFG_HEIGHT_OFFSET);
+	writel(reg, mxc_isi->regs + CHNL_SCL_IMG_CFG);
+
 	/* CSC */
 	mxc_isi_channel_set_csc(mxc_isi);
 
-- 
1.7.9.5

